modules/ximgproc/src/seeds.cpp

```diff
@@ -959,20 +959,22 @@ bool SuperpixelSEEDSImpl::probability(int image_idx, int label1, int label2,
             p = 1.f;
         switch( seeds_prior )
         {
-        case 5: p *= p;
-            //no break
-        case 4: p *= p;
-            //no break
-        case 3: p *= p;
-            //no break
-        case 2:
+        case_2:        case 2:
             p *= p;
             P_label1 *= T[seeds_top_level][label2];
             P_label2 *= T[seeds_top_level][label1];
             //no break
         case 1:
             P_label1 *= p;
             break;
+
+        case 5: p *= p;
+            //no break
+        case 4: p *= p;
+            //no break
+        case 3: p *= p;
+            //no break
+	  goto case_2;
```

Comment:
Is it possible to avoid using goto construction? It harms code readability and maintenance to my opinion